pkg/resolver/search.go

```diff
@@ -34,10 +35,29 @@ type SearchResult struct {
 
 const ErrorMsg string = "Error building Search query:"
 
+// isContextError checks if an error is a context timeout or cancellation error
+func isContextError(err error) bool {
+	if err == nil {
+		return false
+	}
+	errMsg := err.Error()
+	return errors.Is(err, context.DeadlineExceeded) ||
+		errors.Is(err, context.Canceled) ||
+		errMsg == "context deadline exceeded" ||
+		errMsg == "context canceled"
+}
+
 func Search(ctx context.Context, input []*model.SearchInput) ([]*SearchResult, error) {
 	defer metrics.SlowLog("SearchResolver", 0)()
 	// For each input, create a SearchResult resolver.
 	srchResult := make([]*SearchResult, len(input))
+
+	select {
+	case <-ctx.Done():
+		return srchResult, errors.New("context exceeded deadline")
```

Comment:
This error message is sent in the response to the client. Let's make it more readable.
Suggestion: "Timeout processing the request."